### DISCUSSION TITLE
hmac 256

### DISCUSSION THREAD

**Discussion by 3105545** [lsh123/xmlsec#768] [Q&A]
Hi, could you tell me where I can find an example code to get an hmac signature as in \tests\aleksey-xmldsig-01\enveloping-sha256-hmac-sha256.xml

**Comment by lsh123** [comment]
I don't think there are examples of HMAC signatures, but you can take the example with RSA signatures and simply replace the template and key to HMAC:

https://github.com/lsh123/xmlsec/blob/master/examples/sign1.c

**Comment by lsh123** [comment]
Also, you can run tests and the exact xmlsec command line tool commands will be printed in the log file. Then you can follow the logic in the command line tool to recreate the standalone code:

https://github.com/lsh123/xmlsec/blob/master/tests/testDSig.sh#L644
